dbms/src/Storages/getStructureOfRemoteTable.cpp

```diff
@@ -31,9 +32,48 @@ ColumnsDescription getStructureOfRemoteTable(
     const Context & context,
     const ASTPtr & table_func_ptr)
 {
-    /// Send to the first any remote shard.
-    const auto & shard_info = cluster.getAnyShardInfo();
+    const auto & shards_info = cluster.getShardsInfo();
 
+    std::string fail_messages;
```

Comment:
Maybe `hasColumnInTable` function won't work?

Reply:
Why?

Reply:
Sorry I missed it's already in its own test.